connectors/connectors/rest/rest-swagger-connector/src/main/java/io/syndesis/connector/swagger/SwaggerConnectorComponent.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.connector.swagger;
+
+import java.io.FileOutputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.camel.Endpoint;
+import org.apache.camel.component.connector.DefaultConnectorComponent;
+import org.apache.camel.component.connector.DefaultConnectorEndpoint;
+import org.apache.camel.component.rest.swagger.RestSwaggerEndpoint;
+import org.apache.camel.util.IntrospectionSupport;
+import org.apache.camel.util.IntrospectionSupport.ClassInfo;
+import org.apache.commons.io.IOUtils;
+
+public class SwaggerConnectorComponent extends DefaultConnectorComponent {
+
+    private String specification;
+
+    public SwaggerConnectorComponent() {
+        super("swagger-connector", SwaggerConnectorComponent.class.getName());
+    }
+
+    public String getSpecification() {
+        return specification;
+    }
+
+    public void setSpecification(final String specification) {
+        this.specification = specification;
+    }
+
+    @Override
+    protected Endpoint createEndpoint(final String uri, final String remaining, final Map<String, Object> parameters)
+        throws Exception {
+        final URI baseEndpointUri = URI.create(uri);
+
+        final String scheme = Optional.ofNullable(baseEndpointUri.getScheme()).orElse(baseEndpointUri.getPath());
+
+        final String swaggerSpecificationPath = scheme + ".swagger";
+
+        try (final OutputStream out = new FileOutputStream(swaggerSpecificationPath)) {
+            IOUtils.write(specification, out, StandardCharsets.UTF_8);
```

Comment:
This writes always in the current directory. Are you sure that this is always writable ? Wouldn't it be better to store and reference it from a temp directory ?